src/test/java/cloud/fogbow/ras/core/plugins/interoperability/openstack/securityrule/v2/OpenStackSecurityRulesPluginTest.java

```diff
@@ -1,258 +1,512 @@
     private static final String FAKE_PROJECT_ID = "fake-project-id";
-    private static final String FAKE_USER_NAME = "fake-user-name";
-    private static final String FAKE_MEMBER_ID = "fake-member-id";
-    private static final String FAKE_GATEWAY = "fake-gateway";
-    private static final String FAKE_ADDRESS = "fake-address";
-
-    private OpenStackSecurityRulePlugin openStackSecurityRulePlugin;
-    private OpenStackV3User openStackV3Token;
-    private HttpRequestClient clientUtil;
+    private static final String MAP_METHOD = "map";
+    private static final String PROTOCOL_ANY = "ANY";
+    private static final String PROTOCOL_ICMP = "ICMP";
+    private static final String PROTOCOL_TCP = "TCP";
+    private static final String PROTOCOL_UDP = "UDP";
+
+
+    private OpenStackSecurityRulePlugin plugin;
+    private OpenStackV3User cloudUser;
     private Properties properties;
-    private OpenStackHttpClient openStackHttpClient;
+    private OpenStackHttpClient clientMock;
+    private NetworkOrder majorOrder;
 
     @Before
     public void setUp() throws InvalidParameterException, UnexpectedException {
-        PropertiesHolder propertiesHolder = PropertiesHolder.getInstance();
-        this.properties = propertiesHolder.getProperties();
-        this.properties.put(NETWORK_NEUTRONV2_URL_KEY, DEFAULT_NETWORK_URL);
-
+        this.testUtils.mockReadOrdersFromDataBase();
         String confFilePath = HomeDir.getPath() + SystemConstants.CLOUDS_CONFIGURATION_DIRECTORY_NAME + File.separator
                 + "default" + File.separator + SystemConstants.CLOUD_SPECIFICITY_CONF_FILE_NAME;
 
-        this.openStackSecurityRulePlugin = Mockito.spy(new OpenStackSecurityRulePlugin(confFilePath));
+        this.properties = Mockito.mock(Properties.class);
+        Mockito.when(this.properties.get(NETWORK_NEUTRONV2_URL_KEY)).thenReturn(DEFAULT_NETWORK_URL);
 
-        this.clientUtil = Mockito.mock(HttpRequestClient.class);
-        this.openStackHttpClient = Mockito.spy(new OpenStackHttpClient());
-        this.openStackSecurityRulePlugin.setClient(this.openStackHttpClient);
-        this.openStackV3Token = new OpenStackV3User(FAKE_USER_ID, FAKE_NAME, FAKE_TOKEN_VALUE, FAKE_PROJECT_ID);
-    }
+        PowerMockito.mockStatic(PropertiesUtil.class);
+        BDDMockito.when(PropertiesUtil.readProperties(Mockito.eq(confFilePath))).thenReturn(properties);
+
+        this.plugin = Mockito.spy(new OpenStackSecurityRulePlugin(confFilePath));
 
-    @After
-    public void validate() {
-        Mockito.validateMockitoUsage();
+        this.clientMock = Mockito.mock(OpenStackHttpClient.class);
+        this.plugin.setClient(this.clientMock);
+        this.cloudUser = new OpenStackV3User(FAKE_USER_ID, FAKE_NAME, FAKE_TOKEN_VALUE, FAKE_PROJECT_ID);
+
+        this.majorOrder = testUtils.createNetworkOrder(testUtils.getLocalMemberId(), testUtils.FAKE_REMOTE_MEMBER_ID);
     }
 
     // test case: The http client must make only 1 request
```

Comment:
nit: maybe we could improve this comment a little?

Reply:
Okay